apps/Gruntfile.js

```diff
@@ -622,6 +623,113 @@ module.exports = function (grunt) {
     'ejs'
   ]);
 
+  grunt.registerTask('check-entry-points', function () {
+    console.log("checking entry points");
+    let numBadEntryPoints = 0;
+    var validDirectories = {
+      'studio': ['../dashboard/app/views'],
+      'code.org': ['../pegasus/sites.v3/code.org/public'],
+    };
+
+    function walkDirectory(directory, callback) {
+      var files = fs.readdirSync(path.resolve(directory));
+      files.forEach(function (filename) {
+        var fullPath = path.resolve(directory, filename);
+        if (fs.statSync(fullPath).isDirectory()) {
+          walkDirectory(fullPath, callback);
+        } else {
+          callback(fullPath);
+        }
+      });
+    }
+
+    for (var key in config.webpack.build.entry) {
+      var errors = [];
+      var usageCount = 0;
+      var entryPointPath = config.webpack.build.entry[key][1];
+      var searchString = 'js/'+key+'.js';
+      var possibleValidKeys = new Set();
+      var matchedTemplatePaths = {};
+      var entryPointPatternMatch = entryPointPath.match(
+        /^\.\/src\/sites\/([\w.]+)\/pages\/(.*)\.jsx?$/
+      );
+      let directoriesToSearch = [];
+      if (entryPointPatternMatch) {
+        console.log('this is for site', entryPointPatternMatch[1]);
+        directoriesToSearch = validDirectories[entryPointPatternMatch[1]] || [];
+      }
+      directoriesToSearch.forEach(function (directory)  {
+        matchedTemplatePaths[directory+'/'] = [];
+        var fullDirectoryPath = path.resolve(directory);
+        walkDirectory(fullDirectoryPath, function (filePath) {
+          if (fs.readFileSync(filePath).indexOf(searchString) >= 0) {
```

Comment:
I'll give this a shot tomorrow and see how fast it runs.  Is there a reason we shouldn't do this in parallel (at least somewhat)?  Seems like we're only reading, and it might speed things up.